File: python-project-euler-100/p022.py
from Solution import Solution

class P022(Solution):
    def read_names(self, file_name):
        names = []
        with open(file_name) as file_in:
            parts = file_in.read().split(',')
            for part in parts:
                names.append(part[1:-1])
            names.sort()
        return names

    def calculate_name_value(self, name):
        value = 0
        for i in range(len(name)):
            value += (ord(name[i]) - ord('A') + 1)
        return value
    
    def solve(self):
        self.problem_number = 22
        result = 0
        names = self.read_names('p022.txt')

        for i in range(len(names)):
            result += (self.calculate_name_value(names[i]) * (i + 1))

        return result

def main():
	P022().run()

if __name__ == "__main__":
	main()

File: go-project-euler-100/p022.go
package main

import (
	_ "embed"
	"sort"
	"strings"
)

//go:embed p022.txt
var p022Data string

func p022NameValue(name string) int {
	value := 0
	for _, ch := range name {
		value += int(ch-'A') + 1
	}
	return value
}

/*P022Solve solution for p022*/
func P022Solve() int {
	raw := strings.TrimSpace(p022Data)
	parts := strings.Split(raw, ",")
	names := make([]string, len(parts))
	for i, p := range parts {
		names[i] = strings.Trim(p, "\"")
	}
	sort.Strings(names)

	result := 0
	for i, name := range names {
		result += p022NameValue(name) * (i + 1)
	}
	return result
}
